src/doc-printer.js

```diff
@@ -411,6 +416,16 @@ function printDocToString(doc, options) {
       }
     }
   }
+
+  const cursorPlaceholderIndex = out.indexOf(cursor.placeholder);
+  if (cursorPlaceholderIndex >= 0) {
+    const beforeCursor = out.slice(0, cursorPlaceholderIndex).join("");
+    options.cursorOffsetResult = beforeCursor.length;
```

Comment:
That's much cleaner :) Fixed in https://github.com/prettier/prettier/pull/1637/commits/1fa6b4e0ae24ff347ded5d99a907cc77e8fdd569